src/main/java/org/cactoos/text/SubText.java

```diff
@@ -0,0 +1,120 @@
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.text;
+
+import java.io.IOException;
+import org.cactoos.Scalar;
+import org.cactoos.Text;
+
+/**
+ * Extract a substring from a Text.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.9
+ */
+public final class SubText implements Text {
+
+    /**
+     * The text.
+     */
+    private final Text origin;
+
+    /**
+     * The start position in the text.
+     */
+    private final Scalar<Integer> start;
+
+    /**
+     * The end position in the text.
+     */
+    private final Scalar<Integer> end;
+
+    /**
+     * Ctor.
+     * @param text The String
+     * @param start Start position in the text
+     */
+    public SubText(final String text, final int start) {
+        this(new StringAsText(text), start);
+    }
+
+    /**
+     * Ctor.
+     * @param text The String
+     * @param start Start position in the text
+     * @param end End position in the text
+     */
+    public SubText(final String text, final int start, final int end) {
+        this(new StringAsText(text), start, end);
+    }
+
+    /**
+     * Ctor.
+     * @param text The Text
+     * @param start Start position in the text
+     */
+    public SubText(final Text text, final int start) {
+        this(text, () -> start, () -> text.asString().length());
+    }
+
+    /**
+     * Ctor.
+     * @param text The Text
+     * @param start Start position in the text
+     * @param end End position in the text
+     */
+    public SubText(final Text text, final int start, final int end) {
+        this(text, () -> start, () -> end);
+    }
+
+    /**
+     * Ctor.
+     * @param text The Text
+     * @param start Start position in the text
+     * @param end End position in the text
+     */
+    public SubText(final Text text, final Scalar<Integer> start,
+        final Scalar<Integer> end) {
+        this.origin = text;
+        this.start = start;
+        this.end = end;
+    }
+
+    @Override
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    public String asString() throws IOException {
+        try {
+            return this.origin.asString().substring(
+                this.start.asValue(),
+                this.end.asValue()
+            );
+            // @checkstyle IllegalCatchCheck (1 line)
+        } catch (final Exception ex) {
```

Comment:
@yegor256 I'm sorry, but I didn't get it. The problem is the `Exception`? I put it due the methods `this.start.asValue()` and `this.end.asValue()` that throw an `Exception`.